interfaces/builtin/opengl.go

```diff
@@ -80,6 +80,25 @@ const openglConnectedPlugAppArmor = `
 /var/lib/snapd/lib/vulkan/** r,
 /var/lib/snapd/hostfs/usr/share/vulkan/icd.d/*nvidia*.json r,
 
+# Support reading Vulkan configuration from /etc (checked first by loader)
+# Even if empty, Vulkan loader tries these paths and AppArmor denials cause failures
+/etc/vulkan/ r,
+/etc/vulkan/icd.d/ r,
+/etc/vulkan/icd.d/*.json r,
+/etc/vulkan/implicit_layer.d/ r,
+/etc/vulkan/implicit_layer.d/*.json r,
+/etc/vulkan/explicit_layer.d/ r,
+/etc/vulkan/explicit_layer.d/*.json r,
+
+# Comprehensive Vulkan ICD/layer support (includes Mesa/Asahi)
+/usr/share/vulkan/ r,
+/usr/share/vulkan/icd.d/ r,
+/usr/share/vulkan/icd.d/*.json r,
+/usr/share/vulkan/implicit_layer.d/ r,
+/usr/share/vulkan/implicit_layer.d/*.json r,
+/usr/share/vulkan/explicit_layer.d/ r,
+/usr/share/vulkan/explicit_layer.d/*.json r,
```

Comment:
The paths listed here do not exist in the base snaps. Application snaps are expected to use runtimes, e.g. mesa runtime to provide the drivers, which the snap can then place at some specific location using layouts. However, this is not needed most of the time, as the snapcraft provided integration should set up the execution environment such that the locations from mesa snap (or other snap shipping gpu drivers) are automatically included at runtime.